### PR TITLE
DHFPROD-2171: Resolve errors loading online-store in QS

### PR DESCRIPTION
**Pull request by ryanjdew**

### CODE REVIEW

File: examples/online-store/flows/LoadOrders.flow.json [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-    "1": {
-      "stepDefinitionName": "IngestOrders",
-      "stepDefinitionType": "CUSTOM"
+    "1" : {
+      "name" : "IngestOrders",
+      "description" : "null",
+      "options" : { },
+      "customHook" : null,
+      "retryLimit" : 0,
+      "batchSize" : 0,
```
Comment (akshaysonvane) [inline examples/online-store/flows/LoadOrders.flow.json]: Should be null or matching the flow
```suggestion
      "batchSize" : null,
```

File: examples/online-store/flows/LoadOrders.flow.json [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
+      "threadCount" : 0,
```
Comment (akshaysonvane) [inline examples/online-store/flows/LoadOrders.flow.json]: Should be null or matching the flow
```suggestion
      "threadCount" : null,
```

File: marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java [hunk 3]
```diff
@@ -133,15 +133,19 @@ public static Step deserialize(JsonNode json) {
         Step step = new Step();
 
         JSONObject jsonObject = new JSONObject(json);
-        step.setName(jsonObject.getString("name"));
+        step.setStepDefinitionName(jsonObject.getString("stepDefinitionName"));
+        step.setStepDefinitionType(StepDefinition.StepDefinitionType.getStepDefinitionType(jsonObject.getString("stepDefinitionType")));
+        String stepName = jsonObject.getString("name");
+        if (stepName == null) {
+            stepName = step.getStepDefinitionName();
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java]: Is step name optional?

Comment (aebadirad) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java]: Yes, but that's not encouraged.

PKey for the step is the #.

Comment (ryanjdew) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java]: Good question. The backend doesn't require it and I don't believe the middle-tier does either, but Quick Start seems to.

Comment (akshaysonvane) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java]: We seem to track both step name and step-def name in prov.
Should we make it required as going forward when we have a step-def bank the step name will come in handy to track the relation between both?

Comment (aebadirad) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java]: Yeah. Going forward we should.

**Comment by aebadirad** [comment]
Rebase needed

**Pull request merged**